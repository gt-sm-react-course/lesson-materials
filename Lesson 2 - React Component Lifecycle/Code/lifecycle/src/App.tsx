import "./App.css";

//import ParentComponent from "./ClassComponents/ParentComponent.tsx";
import ParentComponent from "./FunctionalComponents/ParentComponent";

const App: React.FC = () => {
  return <ParentComponent />;
};

export default App;
