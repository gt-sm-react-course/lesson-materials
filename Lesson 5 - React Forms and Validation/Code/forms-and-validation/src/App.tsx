import RegistrationFormComponent from "./components/1. RegistrationFrom";
// import RegistrationFormComponent from "./components/2. RegistrationFromValidation";
// import RegistrationFormComponent from "./components/3. RegistrationFromValidationWithAPI";
// import RegistrationFormComponent from "./components/4. RegistrationFromWithFormik";
// import RegistrationFormComponent from "./components/5. RegistrationFromWithFormikAndYup";

function App() {
  return <RegistrationFormComponent />;
}

export default App;
