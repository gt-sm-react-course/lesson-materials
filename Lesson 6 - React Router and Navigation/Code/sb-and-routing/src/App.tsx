// // =============== First Example
// import { BrowserRouter, Route, Routes, useParams } from "react-router-dom";
// import { UserList } from "./components/UserList";
// import { PostList } from "./components/PostList";

// const PostListWrapper = () => {
//   const { userId } = useParams();
//   return <PostList userId={Number(userId)} />;
// };

// function App() {
//   return (
//     <BrowserRouter>
//       <Routes>
//         <Route path="/" element={<UserList />} />
//         <Route path="/:userId/posts" element={<PostListWrapper />} />
//       </Routes>
//     </BrowserRouter>
//   );
// }

// export default App;

// // =============== Second Example
// import { BrowserRouter, Route, Routes } from "react-router-dom";

// import Layout from "./layouts/Layout";
// import NotFoundPage from "./pages/NotFoundPage";
// import ProtectedPage from "./pages/ProtectedPage";
// import UserPostsPage from "./pages/UserPostsPage";
// import UsersPage from "./pages/UsersPage";
// import ProtectedRoute from "./ProtectedRoute";

// function App() {
//   return (
//     <BrowserRouter>
//       <Routes>
//         <Route path="/" element={<Layout />}>
//           <Route index element={<UsersPage />} />
//           <Route path=":userId/posts" element={<UserPostsPage />} />
//           <Route
//             path="/secret"
//             element={
//               <ProtectedRoute>
//                 <ProtectedPage />
//               </ProtectedRoute>
//             }
//           />
//           <Route path="*" element={<NotFoundPage />} />
//         </Route>
//       </Routes>
//     </BrowserRouter>
//   );
// }

// export default App;

// =============== 3 example

import { RouterProvider } from "react-router-dom";
import router from "./router";

function App() {
  return <RouterProvider router={router} />;
}

export default App;
