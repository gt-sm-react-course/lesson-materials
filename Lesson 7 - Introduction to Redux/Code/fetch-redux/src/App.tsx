import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import store, { RootState } from "./store";
import {
  GET_USERS_REQUEST,
  GET_USERS_SUCCESS,
  GET_USERS_FAILURE,
  User,
} from "./types";

const fetchUsers = async (dispatch: React.Dispatch<any>) => {
  dispatch({ type: GET_USERS_REQUEST });
  try {
    const response = await fetch("https://jsonplaceholder.typicode.com/users/");
    const data: User[] = await response.json();
    dispatch({ type: GET_USERS_SUCCESS, payload: data });
    console.log(store.getState());
  } catch (error) {
    dispatch({ type: GET_USERS_FAILURE, payload: error.message });
  }
};

const App: React.FC = () => {
  const dispatch = useDispatch();
  const { users, loading, error } = useSelector(
    (state: RootState) => state.user
  );

  useEffect(() => {
    fetchUsers(dispatch);
  }, [dispatch]);

  if (loading) return <div>Loading...</div>;
  if (error) return <div>Error: {error}</div>;

  return (
    <div>
      {users.map((user) => (
        <div key={user.id}>
          <h3>{user.name}</h3>
          <p>Email: {user.email}</p>
          <p>Website: {user.website}</p>
        </div>
      ))}
    </div>
  );
};

export default App;
