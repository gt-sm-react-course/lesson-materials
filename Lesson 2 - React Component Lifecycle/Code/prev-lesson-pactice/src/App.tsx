import { useEffect, useState } from 'react';
import './App.css'
import data from './data/stocks.json'
import { Stock } from './shared/types/Stock';
import { StockList } from './features/StockList/StockList';

function App() {
  const [stocks, setStocks] = useState<Stock[]>(data);

  useEffect(() => {
    const interval = setInterval(() => {
      const updatedStocks = stocks.map((stock) => {
        const changePercent = Math.random() * 0.08 - 0.04;
        const updatedPrice = stock.price * (1 + changePercent);
        return { ...stock, price: updatedPrice };
      });
      setStocks(updatedStocks);
    }, 1000);

    return () => clearInterval(interval);
  }, [stocks]);

  return <StockList stocks={stocks} />;
}

export default App
