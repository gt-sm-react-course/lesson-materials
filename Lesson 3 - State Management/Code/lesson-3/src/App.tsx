import "./App.css";
// import TaskListComponent from './components/Demo 1'
// import TaskListComponent from './components/Demo 2'
// import TaskListComponent from './components/Demo 3'
import TaskListComponent from './components/Demo 4'

function App() {
  return (
    <TaskListComponent />
  )
}

export default App;
