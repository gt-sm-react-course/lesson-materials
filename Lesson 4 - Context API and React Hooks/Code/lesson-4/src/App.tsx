import "./App.css";

import UsersList from "./components/1. UsersList";
// import UsersList from "./components/2. UsersListWithReducer";
// import UsersList from "./components/3. UsersListWithCustomHook";
// import UsersList from "./components/4. UsersListWithContext";
// import { UserProvider } from "./components/4. UsersListWithContext/UserContext";

function App() {
  return <UsersList />;
}

// function App() {
//   return (
//     <UserProvider>
//       <UsersList />
//     </UserProvider>
//   );
// }

export default App;
