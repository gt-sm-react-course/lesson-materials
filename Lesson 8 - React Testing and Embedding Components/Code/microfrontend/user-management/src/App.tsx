import "./App.css";
import UserList from "./features/UserList";

function App() {
  return (
    <>
      <h1>Component Application</h1>
      <div className="card">
        <UserList />
      </div>
    </>
  );
}

export default App;
