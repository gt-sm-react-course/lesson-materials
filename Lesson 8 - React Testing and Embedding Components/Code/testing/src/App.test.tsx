import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import App from "./App";
import useFetchUsers from "./hooks/useFetchUsers";

jest.mock("./hooks/useFetchUsers");

const mockUseFetchUsers = useFetchUsers as jest.Mock;

describe("App Component", () => {
  beforeEach(() => {
    mockUseFetchUsers.mockReturnValue([
      {
        userId: 1,
        firstname: "John",
        lastname: "Doe",
        email: "john@example.com",
      },
      {
        userId: 2,
        firstname: "Jane",
        lastname: "Doe",
        email: "jane@example.com",
      },
    ]);
  });

  test('App component snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the users list component", () => {
    render(<App />);
    expect(screen.getByText("Users List")).toBeInTheDocument();
    expect(screen.getByText("John")).toBeInTheDocument();
    expect(screen.getByText("Jane")).toBeInTheDocument();
  });

  test("filters users by first name", async () => {
    render(<App />);
    fireEvent.change(screen.getByPlaceholderText("Filter by First Name"), {
      target: { value: "John" },
    });
    fireEvent.click(screen.getByText("Filter"));

    await waitFor(() => {
      expect(screen.queryByText("Jane")).not.toBeInTheDocument();
      expect(screen.getByText("John")).toBeInTheDocument();
    });
  });

  test("filters users by last name", async () => {
    render(<App />);
    fireEvent.change(screen.getByPlaceholderText("Filter by Last Name"), {
      target: { value: "Doe" },
    });
    fireEvent.click(screen.getByText("Filter"));

    await waitFor(() => {
      expect(screen.getByText("John")).toBeInTheDocument();
      expect(screen.getByText("Jane")).toBeInTheDocument();
    });
  });

  test("displays no records message when no users match the filter", async () => {
    render(<App />);
    fireEvent.change(screen.getByPlaceholderText("Filter by Email"), {
      target: { value: "nonexistent@example.com" },
    });
    fireEvent.click(screen.getByText("Filter"));

    await waitFor(() => {
      expect(screen.getByText("No Records")).toBeInTheDocument();
    });
  });
});
