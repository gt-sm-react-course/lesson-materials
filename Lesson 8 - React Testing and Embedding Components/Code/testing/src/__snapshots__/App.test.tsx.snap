// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App component snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      class="text-center my-4"
    >
      Users List
    </h1>
    <div
      class="mb-3"
    >
      <div
        class="input-group"
      >
        <input
          class="form-control"
          placeholder="Filter by First Name"
          type="text"
          value=""
        />
        <input
          class="form-control"
          placeholder="Filter by Last Name"
          type="text"
          value=""
        />
        <input
          class="form-control"
          placeholder="Filter by Email"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary"
        >
          Filter
        </button>
      </div>
    </div>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            First
          </th>
          <th>
            Last
          </th>
          <th>
            Email
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            John
          </td>
          <td>
            Doe
          </td>
          <td>
            john@example.com
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Jane
          </td>
          <td>
            Doe
          </td>
          <td>
            jane@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
